src/main/java/pl/psnc/dei/ui/components/imports/CandidatesListsComponent.java

```diff
@@ -0,0 +1,111 @@
+
+public class CandidatesListsComponent extends VerticalLayout {
+
+	private final CreateImportComponent.FieldFilter recordIdField = (currentRecord, currentValue) -> StringUtils.containsIgnoreCase(currentRecord.getIdentifier(), currentValue);
+	private final CreateImportComponent.FieldFilter recordDatasetField = (currentRecord, currentValue) -> StringUtils.containsIgnoreCase(getDatasetValue(currentRecord), currentValue);
+
+	private ProjectsRepository projectsRepository;
+	private ImportPackageService importPackageService;
+
+	private Set<Record> records;
+
+	private Grid<Record> recordsList;
+	private Button deleteButton;
+	private Project project;
+
+	public CandidatesListsComponent(ProjectsRepository projectsRepository, ImportPackageService importPackageService) {
+		this.projectsRepository = projectsRepository;
+		this.importPackageService = importPackageService;
+		add(createProjectSelection());
+	}
+
+	private void refresh() {
+		records = importPackageService.getCandidates(project.getProjectId(), null);
+		if (recordsList != null) {
+			remove(recordsList);
+		}
+		recordsList = generateGrid();
+		add(recordsList);
+		if (deleteButton != null) {
+			remove(deleteButton);
+		}
+		deleteButton = generateDeleteButton();
+		add(deleteButton);
+	}
+
+	private Button generateDeleteButton() {
+		Button button = new Button("Remove selected records");
+		button.addClickListener(e -> {
+			importPackageService.removeRecordsFromCandidates(recordsList.getSelectedItems());
+			refresh();
+		});
+		return button;
+	}
+
+	private Grid<Record> generateGrid() {
+		Grid<Record> recordsGrid = new Grid<>();
+
+		ListDataProvider<Record> dataProvider = new ListDataProvider<>(records);
+		recordsGrid.setDataProvider(dataProvider);
+		recordsGrid.setSelectionMode(Grid.SelectionMode.MULTI);
+
+		Grid.Column<Record> recordIdColumn = recordsGrid.addColumn(Record::getIdentifier).setHeader("Id").setSortable(true).setFlexGrow(10);
+		Grid.Column<Record> datasetColumn = recordsGrid.addColumn(Record::getDataset).setHeader("Dataset").setSortable(true).setFlexGrow(10);
+
+		HeaderRow filterRow = recordsGrid.appendHeaderRow();
+		addFilter(dataProvider, filterRow, recordIdColumn, recordIdField);
+		addFilter(dataProvider, filterRow, datasetColumn, recordDatasetField);
+
+		recordsGrid.setColumnReorderingAllowed(true);
+		return recordsGrid;
+	}
+
+	private String getDatasetValue(Record record) {
+		return record.getDataset() != null ? record.getDataset().getName() : "";
+	}
+
+	private Component createProjectSelection() {
+		Select<Project> projectSelect = new Select<>();
+		ListDataProvider<Project> listDataProvider = new ListDataProvider<>(projectsRepository.findAll());
+		projectSelect.setDataProvider(listDataProvider);
+
+		projectSelect.addValueChangeListener(event -> {
+			project = event.getValue();
+			refresh();
+		});
+		HorizontalLayout projectSelectionLayout = new HorizontalLayout();
+		projectSelectionLayout.add(new Label("Select project"));
```

Comment:
It probably will look a little bit better if you use select's built-in label.
`projectSelect.setLabel("Select project");`
This way, additional HorizontalLayout is also unnecessary.